pil-analyzer/src/statement_processor.rs

```diff
@@ -643,4 +651,50 @@ where
             functions,
         }
     }
+
+    fn process_trait_implementation(
+        &self,
+        trait_impl: parsed::TraitImplementation<parsed::Expression>,
+    ) -> TraitImplementation<Expression> {
+        let type_vars = trait_impl.type_scheme.vars.vars().collect();
```

Comment:
Since we don't support generic impls, we should add an `unimplemented` assert here that `type_vars` should be empty.